src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -599,13 +599,28 @@ private async Task InternalFetchChildren()
                     //      children of this value can be assumed to alternate between keys and values.'
                     //
                     List<VariableInformation> listChildren = new List<VariableInformation>();
-                    for (int p = 0; p + 1 < children.Length; p += 2)
+                    for (int p = 0; (p + 1) < children.Length; p += 2)
                     {
-                        // One Variable is created for each pair returned with the first element (p) being the name of the child
-                        // and the second element (p+1) becoming the value.
-                        string name = children[p].FindString("value");
-                        var variable = new VariableInformation(children[p + 1], this, '[' + name + ']');
-                        listChildren.Add(variable);
+                        if (children[p].TryFindUint("numchild").GetValueOrDefault(0) > 0)
```

Comment:
nit: `null > 0` will evaluate to false. So you can just remove `.GetValueOrDefault(0)`